esm/index.js

```diff
@@ -51,8 +51,8 @@ const create = block => {
  * @template T
  * @type {<T>(fn: (v?: T) => T | undefined, value?: T) => () => void}
  */
-export const effect = (fn, value) => {
-  const fx = create(() => { value = fn(value) });
+export const effect = (fn) => {
+  let teardown, fx = create(() => { teardown?.call?.(); teardown = fn() });
```

Comment:
where is this returned?

Reply:
It's used as returned value from effect

```js
effect(() => {
  // main effect code, eg.
  let id = setInterval(() => count++, 1000)
  return () => {
    // calls before next effect call
    clearTimeout(id)
  }
})
```

But you're right, I forgot that we should also run teardown when we dispose effect.

Reply:
@WebReflection please have a look